share/p2p/shrexnd/server.go

```diff
@@ -32,7 +32,7 @@ type Server struct {
 }
 
 // NewServer creates new Server
-func NewServer(host host.Host, store *eds.Store, getter *getters.IPLDGetter, opts ...Option) (*Server, error) {
+func NewServer(host host.Host, store *eds.Store, getter share.Getter, opts ...Option) (*Server, error) {
```

Comment:
Just a note that we should make a proxy constructor for this during integration that takes an actual implementation rather than an interface, so we don't accidentally end up with a Server that requests data instead of returning not found. Honestly, keeping the specific implementation there feels safer

Reply:
causes a cyclical dependency when shrexgetter is introduced, so proxy is better here for now